src/frontend/src/lib/utils/token.utils.ts

```diff
@@ -183,3 +185,22 @@ export const sumUsdBalances = ([usdBalance1, usdBalance2]: [
  */
 export const isRequiredTokenWithLinkedData = (token: Token): token is RequiredTokenWithLinkedData =>
 	'twinTokenSymbol' in token && typeof token.twinTokenSymbol === 'string';
+
+/** Find in the provided tokens list a twin IC token by symbol.
+ *
+ * @param tokenToPair - token to find a twin for.
+ * @param tokens - The list of tokens.
+ * @returns IcCkToken or undefined if no twin token found.
+ * */
+export const findTwinToken = ({
+	tokenToPair,
+	tokens
+}: {
+	tokenToPair: Token;
+	tokens: Token[];
+}): IcCkToken | undefined =>
+	isRequiredTokenWithLinkedData(tokenToPair)
+		? (tokens.find(
+				(token) => token.symbol === tokenToPair.twinTokenSymbol && isIcCkToken(token)
+			) as IcCkToken | undefined)
```

Comment:
shouldn't `isIcCkToken` already cast it as `IcCkToken`? is the casting necessary?

`export const isIcCkToken = (token: Token): token is IcCkToken =>...`

Reply:
Unfortunately, still necessary.